### PR TITLE
Refactor security service

### PR DESCRIPTION
**Pull request by danvixent**

### CODE REVIEW

File: api/portal-api/portal_api.go [hunk 1]
```diff
@@ -129,7 +129,7 @@ func (a *PortalLinkHandler) retrievePortalLink(r *http.Request) (*datastore.Port
 
 	if util.IsStringEmpty(token) {
 		cred, _ := middleware.GetAuthFromRequest(r)
-		token = cred.Token
+		token = cred.Token // TODO(all): this should probably be cred.APIKey, token is used for jwt see  middleware.GetAuthFromRequest(r)
```
Comment (jirevwe) [inline api/portal-api/portal_api.go:132]: Hmm, does this mean that we would need to pass the token in a different header?

File: services/create_api_key.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	}
+
+	dk := pbkdf2.Key([]byte(key), []byte(salt), 4096, 32, sha256.New)
+	encodedKey := base64.URLEncoding.EncodeToString(dk)
+
+	apiKey := &datastore.APIKey{
+		UID:       ulid.Make().String(),
+		MaskID:    maskID,
+		Name:      ss.NewApiKey.Name,
+		Type:      ss.NewApiKey.Type, // TODO: this should be set to datastore.ProjectKey
```
Comment (jirevwe) [inline services/create_api_key.go:74]: is this todo still valid?

**Review by jirevwe — CHANGES_REQUESTED** [review]
It's fine, I just need info about those comments

**Review by Dotunj — APPROVED** [review]

File: api/portal-api/portal_api.go (hunk 1 above)
Comment (danvixent) [reply api/portal-api/portal_api.go:132]: No, we found this fix.

File: services/create_api_key.go (hunk 2 above)
Comment (danvixent) [reply services/create_api_key.go]: No, i've removed it

**Review by jirevwe — APPROVED** [review]

**Pull request merged**